src/Fabric/init.spec.lua

```diff
@@ -228,5 +248,27 @@ return function()
 			expect(component:isDestroyed()).to.equal(true)
 			expect(otherComoponent:isDestroyed()).to.equal(false)
 		end)
+
+		it("shouldn't listen before registering", function()
+			local newComponent = {
+				name = "Test2",
+				tag = "Test2",
+			}
+			local newComponent2 = {
+				name = "Test3",
+				tag = "Test3",
+				components = {
+					Test2 = {}
+				},
+			}
+			fabric:registerComponent(newComponent2)
+			CollectionService:AddTag(testInstance, "Test3")
+			expect(CollectionService:HasTag(testInstance, "Test3")).to.equal(true)
+			fabric:registerComponent(newComponent)
+			heartbeat()
+			local component = fabric:getComponentByRef("Test3", testInstance)
+			expect(component).to.be.ok()
+			expect(fabric:getComponentByRef("Test2", component)).to.be.ok()
+		end)
```

Comment:
Can we move this into its own .spec file, called TagWatcher.spec.lua? It should be mostly the same. It should use its own fabric instance with its own namespace.